src/target/riscv/riscv-013.c

```diff
@@ -3592,32 +3656,42 @@ static int read_memory_abstract(struct target *target, target_addr_t address,
 			}
 		}
 
+		/* TODO:
+		 * (1) Only the 1st access can result in a 'skip'
+		 * (2) Analyze cmderr value */
 		if (result != ERROR_OK)
-			return result;
+			return MEM_ACCESS_STATUS_SKIPPED_ABSTRACT_ACCESS_CMDERR;
 
 		/* Copy arg0 to buffer (rounded width up to nearest 32) */
 		riscv_reg_t value;
 		result = read_abstract_arg(target, &value, 0, width32);
 		if (result != ERROR_OK)
-			return result;
+			return MEM_ACCESS_STATUS_FAILED_ABSTRACT_ARG_READ_FAILED;
 		buf_set_u64(p, 0, 8 * size, value);
 
 		if (info->has_aampostincrement == YNM_YES)
 			updateaddr = false;
 		p += size;
 	}
 
-	return result;
+	return (result == ERROR_OK) ? MEM_ACCESS_STATUS_OK : MEM_ACCESS_STATUS_FAILED;
```

Comment:
It looks like the check for `result == ERROR_OK` is redundant.

If I read the code correctly, the result cannot be different than ERROR_OK when we reach the line 3677.

Reply:
Addressed